src/ucp/api/ucp.h

```diff
@@ -439,6 +440,25 @@ enum {
 };
 
 
+/**
+ * @ingroup UCP_MEM
+ * @brief UCP memory mapping protection mode.
+ *
+ * The enumeration list describes the memory mapping protections supported by the @ref
+ * ucp_mem_map() function.
```

Comment:
If underlaying hardware does not support requested access protection level, it can be ignored. @tonycurtis not sure what is the best place to put this. In `ucp_mem_map` or here. Essentially this is a hint.

Reply:
What is the use case for this addition?  Would like a little more context.

Reply:
These are permission flags for memory access: LOCAL/REMOTE-READ/WRITE.

Reply:
Currently PROT_READ | PROT_WRITE is hard code in UCT level(please refer to `create.protection =  PROT_READ | PROT_WRITE` in `uct_knem_mem_reg_internal()`). This cause failure when the application assgin REDA ONLY memory to ucp_mem_map()(refer to bug #4924). So we want to add these flags on UCP level to allow application to set the protection modes, but not hard code in UCT.
Originally we were going to add a prot parameter to `ucp_mem_map`, but this cause effect too much on exist applications.